### PR TITLE
Fix issue of content app losing connection to the platform-app.

### LINKED ISSUE #23535: [BUG] Example tv-app android : content-app loses service binding and is unable to recover.
### Reproduction steps

```bash
1. Run the platform-app and content-app from the tv-app android example.
2. Send a attribute update from the content-app to platform-app(using UI).
3. Kill the platform-app and start it again.
4. Send a attribute update from the content-app to platform-app(using UI).
The attribute update in the last step does not reach the platform-app since the service binding is lost.
```

### Bug prevalence

1-2 time a day when working on the tv-app

### GitHub hash of the SDK that was being used

 b278e34

### Platform

android

### Platform Version(s)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by amitnj**
`Fixes #23535`

Fixed the handling of how the service binding is maintained in the client. Taken care of service disconnections correctly. Maintaining internal state of the client correctly.

### CODE REVIEW

File: examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java [hunk 1]
```diff
@@ -1,13 +1,92 @@
...
 import java.util.HashMap;
 import java.util.Map;
 
 /** Class to hold attribute values to help test attribute read and subscribe use cases. */
 public class AttributeHolder {
-  private static AttributeHolder instance = new AttributeHolder();
   private Map<Long, Map<Long, Object>> attributeValues = new HashMap<>();
+  private static final String TAG = "AttributeHolder";
+
+  public static final String TL_LONG =
```
Comment (chrisdecenzo) [inline examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java:13]: is this just dummy response data for the example content app?

**Review by chrisdecenzo — APPROVED** [review]

File: examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java (hunk 1 above)
Comment (amitnj) [reply examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java:13]: Yes. And I put a few different to test simple case, chunking and error case for message that is too long.

Comment (andy31415) [inline examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java:13]: There should be a comment saying what TL_LONG/SHORT/BAD are. what is a `TL` ?

Comment (amitnj) [reply examples/tv-app/android/App/content-app/src/main/java/com/example/contentapp/AttributeHolder.java:21]: Added comment.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**